GOVERNANCE.md

```diff
@@ -0,0 +1,126 @@
+
+# Roles 
+
+## Contributor
+
+* Definition: a user who has written/commented at least one issue, worked to label/triage issues, written a blog post, given a talk, etc. 
+* How this role is recognized: there is no central list of contributors / no formal recognition for contributors.
+
+## Project member
+
+* Definition: some one who has submitted at least one PR with code contributions, that has been merged into master. (spelling fixes in docs do not count)
+* How to obtain this role: any user can become a member by submitting a PR, then having it reviewed and merged into master. Contributors who have written issues should be encouraged to submit their first PR to become a project member.
+* How this role is recognized: Members are credited via role="ctb" in DESCRIPTION (so they appear in Author list on CRAN), and they are added to https://github.com/orgs/Rdatatable/teams/project-members so they can create new branches in the Rdatatable/data.table GitHub repo. They also appear on https://github.com/Rdatatable/data.table/graphs/contributors (Contributions to master, excluding merge commits).
+
+## Reviewer
+
+* Definition: a member who has volunteered to do code reviews for some features/files. 
+* How to obtain this role: after one or more significant PRs to a given file, a member should be invited to add their name as a reviewer of that file in CODEOWNERS, and after that is merged into master, then they are considered a reviewer. 
+* How this role is recognized: same credit in DESCRIPTION as a regular member, role="ctb" (so they appear in Author list on CRAN). 
+* Note: having your name in CODEOWNERS does not give any special permission, but it does mean that you will be notified whenever there is a new PR with changes to that file. 
+
+## Committer
+
+* Definition: permission to commit to, and merge PRs into, master branch. 
+* How to obtain this role: after a reviewer has a consistent history of careful reviews of others' PRs, then a current Committer should ask all other current Committers if they approve promoting the Reviewer to Committer, and it should be done if there is Consensus among active Committers. 
+* How this role is recognized: credited via role="aut" in DESCRIPTION (so they appear in Author list on CRAN), and added to https://github.com/orgs/Rdatatable/teams/maintainers which gives permission to merge PRs into master branch. 
+
+## CRAN maintainer
+
+* Definition: in charge of communication with CRAN. Responsible for submitting releases to CRAN on a regular basis, and for responding to requests from CRAN.
+* How to obtain this role: (1) merge into master a PR adding role="cre" to DESCRIPTION, and (2) submit updated package to CRAN (previous CRAN maintainer will have to confirm change by email to CRAN). 
+* How this role is recognized: credited via role="cre" in DESCRIPTION, so they appear as Maintainer on CRAN. 
+
+## Other roles
+
+Each of the below roles is important, and a wiki page should be
```

Comment:
Why "should be"? Shouldn't we at least create a stub Wiki before merging this PR?

Reply:
I am against keeping info in wiki, in favor of having extra markdown file in .dev dir, or wherever that document fits

Reply:
I have not carefully thought about this section before writing it, sorry.
I wanted to mention these other roles as important, but I did not want to give full definition / documentation of them in the governance document, so I deferred to the wiki, for example https://github.com/Rdatatable/data.table/wiki/Release-management-and-revdep-checks for the rev dep manager. Do you think I should just remove this section from the governance? or move to markdown file in .dev?

Reply:
I would just cut off this

> and a wiki page should be created to explain (1) details of responsibilities, (2) who to contact
to obtain this role.

As long as those roles are being executed by volunteers it make sense to have them rather loosely defined. Otherwise it imposes pressure what someone "should" be doing.

Reply:
so what change would you suggest? I'm not sure what cut off this means.
I'm leaning toward removing this section entirely, and just mentioning these roles on https://github.com/Rdatatable/data.table/wiki/Contributing

Reply:
cut off = remove

Reply:
removed